### PR TITLE
refactor: Use ComponentObjectPropsOptions helper for type-safety on Props

### PR DESCRIPTION
**Pull request by nickmessing**
## Description
Adds additional type safety by binding Props type declaration to PropTypes object

## Motivation and Context
Vue 3 provides mechanisms to connect (or infer) Props type & PropTypes object. Using it results in increased type safety without any drawbacks.

## How Has This Been Tested?
none

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/effects/elevation.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Setup
-import { computed } from 'vue'
+import { computed, ComponentObjectPropsOptions } from 'vue'
```
Comment (johnleider) [inline packages/vuetify/src/effects/elevation.ts]: Multi line when importing more than 1

**Comment by KaelWD** [comment]
This does cause errors on some stuff in elevation.spec.ts, and "go to definition" now goes to the interface instead of the implementation.

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/contributors @vuetifyjs/core-team

**Comment by johnleider** [comment]
As we are about to begin full v3 development, there are a lot of items in flux. I'll say for now let's close this and revisit once we have a solid foundation.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Pull request closed without merging**